tests/filecheck/backend/riscv/optimisation_riscv.mlir

```diff
@@ -0,0 +1,33 @@
+// RUN: xdsl-opt -p optimise-riscv %s | filecheck %s
+
+builtin.module {
+  "riscv_func.func"() ({
```

Comment:
Let's simplify this test to be as specific as possible about the changes? Maybe something like this:
```suggestion
    %0, %1 = "test.op"() : () -> (!riscv.reg<a0>, !riscv.reg<a1>)
    %2 = riscv.mv %0 : (!riscv.reg<a0>) -> !riscv.reg<a0>
    %3 = riscv.mv %1 : (!riscv.reg<a1>) -> !riscv.reg<a2>
```

Reply:
Minimised in the newest commit. Thanks!